frontend/src/pages/GamePage.tsx

```diff
@@ -39,59 +40,105 @@ const GamePage: React.FC = () => {
                     <TerminalOutput key={`t-${index}`}>{`${index + 1}. ${option.text}`}</TerminalOutput>
                 )),
             ]);
         } else {
             setTerminalLineData((prevData) => [
                 ...prevData,
                 <TerminalOutput>
-                    {`Game Over! Final stats - Health: ${stats.health}, Happiness: ${stats.happiness}, Money: ${stats.money}.`}
+                    {`Game Over! Final stats - Health: ${stats.health}, Happiness: ${stats.happiness}, Wealth: ${stats.wealth}.`}
                 </TerminalOutput>,
             ]);
+            setCurrentQuestionIndex(0);
         }
     };
 
+    useEffect(() => {
+        setLoading(true);
+        axios
+            .get("http://localhost:8080/api/getAllQuestions")
+            .then((res) => {
+                const questions = res.data;
+
+                // Ensure questions are an array and map them properly
+                if (!Array.isArray(questions)) {
+                    throw new Error("Invalid response format");
+                }
+
+                setQuizQuestions(questions);
+
+                // Display the first question
+                if (questions.length > 0) {
+                    const firstQuestion = questions[0];
+                    setTerminalLineData((prevData) => [
+                        <TerminalOutput>Welcome to the Life Choices Game!</TerminalOutput>,
+                        <br />,
+                        <TerminalOutput>Your initial stats: Health: 50, Happiness: 50, Wealth: 50.</TerminalOutput>,
+                        <br />,
+                        <TerminalOutput>{`Question 1: ${firstQuestion.content}`}</TerminalOutput>,
+                        ...firstQuestion.options.map((option: any, index: number) => (
+                            <TerminalOutput key={`tout-${index}`}>{`${index + 1}. ${option.text}`}</TerminalOutput>
+                        )),
+                    ]);
+                }
+                setLoading(false);
+            })
+            .catch((error) => {
+                console.error("Error fetching questions:", error);
+                setTerminalLineData((prevData) => [
+                    ...prevData,
+                    <TerminalOutput>Error fetching questions. Please try again later.</TerminalOutput>,
+                ]);
+                setLoading(false);
+            });
+    }, []);
+
     return (
         <div className="container">
-            <Terminal
-                name="Life Choices Game"
-                colorMode={ColorMode.Dark}
-                onInput={handleInput}
-            >
-                {terminalLineData}
-            </Terminal>
+            {loading ? (
+                <p>Loading...</p>
+            ) : (
+                <Terminal
+                    name="Life Choices Game"
+                    colorMode={ColorMode.Dark}
+                    onInput={handleInput}
+                >
+                    {terminalLineData}
```

Comment:
Move the loading component here, because if loading is rendered above this component, it won't be displayed on the terminal. We want everything to be on the terminal.